### PR TITLE
format: Preserve brackets around set union operation

### PR DESCRIPTION
**Pull request by ashutosh-narkar**
This fix preserves the brackets around set union operations when the inflix operator (`|`) is used. Previously the formatter would remove the brackets around the union operation, interpreting the expression as a comprehesion which would yeild unexpected results.

Fixes: #6588

### CODE REVIEW

File: format/format.go [hunk 1]
```diff
@@ -1108,13 +1108,29 @@ func (w *writer) objectWriter() entryWriter {
 		entry := x.([2]*ast.Term)
 		comments = w.writeTerm(entry[0], comments)
 		w.write(": ")
+
+		call, isCall := entry[1].Value.(ast.Call)
+		if isCall && ast.Or.Ref().Equal(call[0].Value) && entry[1].Location.Text[0] == 40 { // Starts with "("
```
Comment (johanfylling) [inline format/format.go:1113]: This should also cover when the call is spread over multiple lines, right?
E.g.: 
```rego
union_object := {"response": (
  declare3 | declare4
)}
```
Let's add that case to the test.

File: format/format.go [hunk 2]
```diff
@@ -1108,13 +1108,29 @@ func (w *writer) objectWriter() entryWriter {
 		entry := x.([2]*ast.Term)
 		comments = w.writeTerm(entry[0], comments)
```
Comment (johanfylling) [inline format/format.go:1109]: What about when the union is in the object key?
E.g.:
```rego
union_object := {(declare3 | declare4): "foo"}
```

File: format/testfiles/test.rego [hunk 3]
```diff
@@ -195,6 +195,13 @@ declare1 := 1
 
 declare2 := 2 { false }
 
+declare3 := {1,2,3}
+declare4 := {4,5,6}
+
+union_object := {"response": (declare3|declare4)}
+
+union_set := {(declare3|declare4)}
```
[the remark this answers is not in the document]
Comment (johanfylling) [reply format/testfiles/test.rego:203]: Another edge case is when multiple, redundant sets of parenthesis are used, e.g.:
```rego
union_set := {(((declare3|declare4)))}
``` 
Should we add another case to assert that we still strip all but one set of parenthesis?

**Review by johanfylling — DISMISSED** [review]
Looks good 👍 .

We might still need to handle the edge case where the union is in the object key, though.

File: format/testfiles/test.rego (hunk 3 above)
[the remark this answers is not in the document]
Comment (ashutosh-narkar) [reply format/testfiles/test.rego:203]: > Should we add a list entry too? 

Done.

> Should we add another case to assert that we still strip all but one set of parenthesis?

Added a test case

File: format/format.go (hunk 1 above)
Comment (ashutosh-narkar) [reply format/format.go:1126]: Yes. Added a test case

File: format/format.go (hunk 2 above)
Comment (ashutosh-narkar) [reply format/format.go:1118]: Updated to handle this case.

**Review by johanfylling — APPROVED** [review]
LGTM 👍
